### DISCUSSION TITLE
API rate limit exceeded on actions/setup-go

### DISCUSSION THREAD

**Discussion by treeder** [community/community#25431] [Actions]
<p>I’m getting an <code>[error]API rate limit exceeded for 199.7.166.17</code> on actions setup. I’m guessing this might apply to everyone from that build box. This is using macOS-latest environment. </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/f/f028ba86a52351e906fd744fea2bd26388aa1d65.png" data-download-href="/uploads/short-url/ygxQV6ZbMJZxw3Fz8tG9kRlARuZ.png?dl=1" title="Screenshot 2019-09-12 at 2.45.53 PM.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/75826/181091085-02863141-9b26-4546-b085-6febcd4a3c5e.png" alt="Screenshot 2019-09-12 at 2.45.53 PM.png" data-base62-sha1="ygxQV6ZbMJZxw3Fz8tG9kRlARuZ" width="690" height="101" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/f/f028ba86a52351e906fd744fea2bd26388aa1d65_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot 2019-09-12 at 2.45.53 PM.png</span><span class="informations">869×128 10.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by francisfuzz** [comment]
<p><a class="mention" href="/treeder">@treeder</a> - Thanks for bringing this up here. I’ve opened an internal issue for our Actions engineers to review this further!</p>

**Reply by Devendranathpvss** [reply]
API rate limit exceeded for user ID XXXXXX. If you reach out to GitHub Support for help, please include the request ID 8800:2DFA:5CA38E3:60E1DEE:659672D1.',

Could you please check and advise us the resolution steps for this kind of issue.

**Comment by francisfuzz** [comment, marked as answer]
<p><a class="mention" href="/treeder">@treeder</a> <a href="https://github.com/actions/setup-go/pull/20" rel="nofollow noopener">actions/setup-go#20</a> just went out that should resolve this issue. I hope this helps!</p>

**Comment by treeder** [comment]
<p>Seems to get past that one, thanks! BUT… we then had a follow up API rate limit exceeded issue trying to use the GitHub API which I’m assuming is a similar issue where all requests coming from the same Actions build machine look like they are coming from one user. Adding some of our own credentials to the API calls fixed that problem though so we worked around it, but I’m sure others will hit that issue. </p>
<p>Again, only seems to happen from Mac builds. </p>

**Comment by DrSensor** [comment]
<p>Just bumped into this issue this day using my own action (not setup-go).</p>
<p>I have the same error when <code>runs-on: macos-latest</code> but not with <code>ubuntu-latest</code>.</p>

**Reply by tlebon** [reply]
> Just bumped into this issue this day using my own action (not setup-go).
> 
> I have the same error when `runs-on: macos-latest` but not with `ubuntu-latest`.

this helped me discover (I believe) the cause of the issue on my side. A github action (but not the one throwing this error) was out of date and running on a deprecated version of node (12-alpine). Updating that action to the newest version made all my frozen dependabot prs finally pass and succeed.

**Comment by asymmetric** [comment]
<p>Same here, this is happening on a macOS builder that does a <code>curl</code>, resulting in</p>
<pre><code class="lang-auto">{"message":"API rate limit exceeded for 199.7.166.17. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)","documentation_url":"https://developer.github.com/v3/#rate-limiting"}
</code></pre>
<p>The same curl on <code>ubuntu-latest</code> succeeds.</p>

**Comment by Ginxo** [comment]
<p>same here <a href="https://github.com/kiegroup/kie-wb-common/pull/3463/checks?check_run_id=1290570196" rel="noopener nofollow ugc">https://github.com/kiegroup/kie-wb-common/pull/3463/checks?check_run_id=1290570196</a></p>
<pre><code class="lang-auto">Error:   API rate limit exceeded for installation ID 3775476.
</code></pre>

**Comment by anilb1010** [comment]
hi . Error: API rate limit exceeded for installation ID 45626221. If you reach out to GitHub Support for help, please include the request ID E401:6728:457375:8CEAC2:65C897C7. 

Would you help me with this topic?
